asv_bench/benchmarks/benchmarks_hindcast.py

```diff
@@ -0,0 +1,166 @@
+                'lead': leads,
+            },
+            dims=('member', 'init', 'lead', 'lon', 'lat'),
+            name='tos',
+            encoding=None,
+            attrs={'units': 'foo units', 'description': 'a description'},
+        )
+        self.reference['tos'] = xr.DataArray(
+            randn((self.ninit, self.nx, self.ny), frac_nan=frac_nan),
+            coords={'lon': lons, 'lat': lats, 'time': inits},
+            dims=('time', 'lon', 'lat'),
+            name='tos',
+            encoding=None,
+            attrs={'units': 'foo units', 'description': 'a description'},
+        )
+
+        self.hist['tos'] = xr.DataArray(
+            randn((self.ninit, self.nx, self.ny, self.nmember), frac_nan=frac_nan),
+            coords={'lon': lons, 'lat': lats, 'time': inits, 'member': members},
+            dims=('time', 'lon', 'lat', 'member'),
+            name='tos',
+            encoding=None,
+            attrs={'units': 'foo units', 'description': 'a description'},
+        )
+
+        self.hind.attrs = {'history': 'created for xarray benchmarking'}
+
+
+class Compute(Generate):
+    """
+    A few examples that benchmark climpred compute_hindcast.
+    """
+
+    def setup(self, *args, **kwargs):
+        self.make_hind_ref()
+
+    @parameterized(['metric', 'comparison'], (METRICS, HINDCAST_COMPARISONS))
+    def time_compute_hindcast(self, metric, comparison):
+        """Take time for compute_hindcast."""
+        _ensure_loaded(
+            compute_hindcast(
+                self.hind, self.reference, metric=metric, comparison=comparison
+            )
+        )
+
+    @parameterized(['metric', 'comparison'], (METRICS, HINDCAST_COMPARISONS))
+    def peakmem_compute_hindcast(self, metric, comparison):
+        """Take memory peak for compute_hindcast for all comparisons."""
+        _ensure_loaded(
+            compute_hindcast(
+                self.hind, self.reference, metric=metric, comparison=comparison
+            )
+        )
+
+    @parameterized(['metric', 'comparison'], (METRICS, HINDCAST_COMPARISONS))
+    def time_bootstrap_hindcast(self, metric, comparison):
+        """Take time for bootstrap_hindcast for one metric."""
+        _ensure_loaded(
+            bootstrap_hindcast(
+                self.hind,
+                self.reference,
+                self.hist,
+                metric=metric,
+                comparison=comparison,
+                bootstrap=bootstrap,
+            )
+        )
+
+    @parameterized(['metric', 'comparison'], (METRICS, HINDCAST_COMPARISONS))
+    def peakmem_bootstrap_hindcast(self, metric, comparison):
+        """Take memory peak for bootstrap_hindcast."""
+        _ensure_loaded(
+            bootstrap_hindcast(
+                self.hind,
+                self.reference,
+                self.hist,
+                metric=metric,
+                comparison=comparison,
+                bootstrap=bootstrap,
+            )
+        )
+
+
+class ComputeDask(Compute):
```

Comment:
Is this here for future use? Currently there are no functions under it.

Reply:
Oh I noticed on my tests that this ran. Must be the `super()` or something else magic in the package. Can you add a quick docstring explaining that it runs all the above tests, but with chunked datasets?